### PR TITLE
W-11934047-connected-apps-revamp

### PR DESCRIPTION
**Pull request by sarathecoubian**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 1]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+NOTE: Only developers who have the organization administrator permission can create apps and specify access scopes in Anypoint Platform.
 
-== Token Flow
+== Create a Connected App
 
-When a developer creates an app, the app requests access to user data.
-When an end user receives such a request and authorizes the app to access their data, the app receives an access token, which allows the app to call Anypoint Platform on the user’s behalf.
-The access token allows the app to see who the user is by using the user’s Anypoint Platform credentials. A token is valid for up to 90 minutes.
+* Your organization can own up to 2000 connected apps, and each connected app can have up to 1000 scopes.
+* Your app is owned by your organization; any organization administrator can control it.
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
* Your app is owned by your organization. Any organization administrator can control it.
```

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 2]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+* A unique `clientID` is automatically assigned and can’t be modified. The `clientSecret` has no expiration date by default but can be modified.
 
-== Endpoint Scopes
+You should also determine the use-case on whether the app is internal to your company or is a third-party application that can be distributed outside of your organization. To create your app, follow these steps:
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
You should also determine whether the app is internal to your company or is a third-party app that can be distributed outside of your organization. To create your app, follow these steps:
```

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 3]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+Note: Additional information may be requested based on the grant type you select. For more information on grant types, see xref:connected-apps-developers.adoc#grant-types-and-token-flow[Grant types and token flow].
++
+.. *Website URL* +
+Users can visit this URL to learn more about the app.
++
+.. *Redirect URIs* +
+Configure which URIs users should be directed to after authorization. Type the URL and click *Add* to add it to the list.
++
+.. *Scopes* +
+The level of permissions to be provided to the app. Click on the *Add Scopes* button to select the scopes. For more information, see xref:connected-apps-developers.adoc#endpoint-scopes[Endpoint Scopes].
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
The level of permissions to provide to the app. Click *Add Scopes* to select the scopes. For more information, see xref:connected-apps-developers.adoc#endpoint-scopes[Endpoint Scopes].
```

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 4]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+. Click *Save*
 
 [NOTE]
-Apps that use `client_credentials` grant type always have the `profile` scope assigned to them implicitly. The data returned by queries using the scope correspond to the user who created the connected app.
+====
+The connected app ownership gets transferred to the root organization owner
+when the user who created the connected app:
 
-For more information on scopes and permissions in the Connected Apps feature, see the scopes documentation page in Access Management:
+* is deleted
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
* is deleted.
```

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 5]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+* is removed from the root organization
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
* is removed from the root organization.
```

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 6]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+* loses administrative privileges over the organization
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
* loses administrative privileges over the organization.
```

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 7]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+
+By clicking on the app name in Access Management > *Connected Apps* > *Owned Apps*, organization administrators can edit the
+owner of the connected app and transfer the ownership to other administrators.
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]: Organization administrators can edit the owner of the connected app and transfer the ownership to other administrators. To do so, click the app name in Access Management > *Connected Apps* > *Owned Apps* and make the appropriate edits.

Comment (IsaacEldridge) [reply modules/ROOT/pages/connected-apps-developers.adoc]: that is just me being picky ^

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 8]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+====
+
+=== Modify the Client Secret for a Connected App
 
 include::partial$include-nav-connected-apps.adoc[]
 
-. Click the *Scopes Documentation* tab. +
-The *Scopes Documentation* page appears, displaying scopes, their respective permissions, and the products to which they belong. 
+. Click on the connected app from the list to enable Edit mode.
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
. Click a connected app from the list to enable Edit mode.
```

Comment (IsaacEldridge) [reply modules/ROOT/pages/connected-apps-developers.adoc]: Or Select an app...?

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 9]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+. The client secret of the connected app
+
+[[application-allowlist-dev]]
+== Application Allowlist
 
-== OIDC Endpoints and Tokens
+Users with the Organization Administrator permission can enable an allowlist to control which apps can access their users' data.
+When your app is added to an allowlist, only the set of scopes associated with the app are reviewed and approved by the administrator.
+If you add new scopes to your app at a later date, you are unable to request those scopes from users until the admin re-approves your app.
+Existing authorizations and new authorizations using the already-allowlisted set of scopes continue to work.
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
Existing authorizations and new authorizations using the previously allowlisted set of scopes continue to work.
```

Comment (IsaacEldridge) [reply modules/ROOT/pages/connected-apps-developers.adoc]: OR: Existing authorizations and new authorizations using the set scopes from the previous (original?) allowlist continue to work.

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 10]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+[[grant-types-and-token-flow]]
+== Grant Types and Token Flow
+
+OAuth framework specification describes several grant types for acquiring access tokens for different use cases.
+Anypoint Platform supports `authorization_code`, `refresh_token`, `password`,  `jwt_bearer`, and `client_credentials` grant types.
+For more information and examples on how to use each grant type, see the https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/access-management-api/minor/1.0/pages/Connected%20App%20Examples/[Connected App Examples] in the Access Management API.
+
+=== Authorization Code
+
+The `authorization_code` grant type is the commonly used grant type to authorize the app to access user’s data. A sample flow for the authorization code grant type could be as follows:
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
The `authorization_code` grant type is commonly used to authorize the app to access a user's data. A sample flow for the authorization code grant type could be as follows:
```

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 11]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
+
+. A developer creates the connected app that acts on behalf of a user with authorization_code grant type.
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
. A developer creates the connected app that acts on behalf of a user with the `authorization_code` grant type.
```

File: modules/ROOT/pages/connected-apps-developers.adoc [hunk 12]
```diff
@@ -16,93 +16,183 @@ All scopes in the endpoint map directly to permissions available in Anypoint Pla
...
-grant_type=refresh_token&client_id=<your_client_id>&client_secret=<your_client_secret>&refresh_token=<your_refresh_token>
-----
+The `password` grant type is used when the application exchanges the user’s username and password for an access token.
+Because the app has to collect the user’s credentials and its limitation with no mechanisms for things like multi-factor authentication, it is not recommended to use this grant type.
 
-Assuming the token request is fulfilled, the refresh token received in the endpoint response is similar to a regular token response, but it might not contain an `id_token`.
+=== JWT Bearer
 
-== Log in with Anypoint Platform Widget
+The `jwt_bearer` (`urn:ietf:params:oauth:grant-type:jwt-bearer`) grant type is used when the application wants to receive access tokens without transmitting sensitive information such as the clientSecret.
```
Comment (IsaacEldridge) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
The `jwt_bearer` (`urn:ietf:params:oauth:grant-type:jwt-bearer`) grant type is used when the application wants to receive access tokens without transmitting sensitive information, such as the clientSecret.
```

File: modules/ROOT/pages/connected-apps-developers.adoc (hunk 7 above)
Comment (sarathecoubian) [inline modules/ROOT/pages/connected-apps-developers.adoc]:
```suggestion
Organization administrators can edit the owner of the connected app and transfer the ownership to other administrators. To do so, click the app name in Access Management > *Connected Apps* > *Owned Apps* and make the appropriate edits.
```

**Review by IsaacEldridge — APPROVED** [review]

**Pull request merged**